dev/AppLifecycle/ValueMarshaling.h

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#pragma once
+
+namespace winrt::Microsoft::Windows::AppLifecycle::implementation
+{
+    static LPCWSTR c_launchSchemeName{ L"ms-launch" };
+    static LPCWSTR c_encodedLaunchSchemeName{ L"ms-encodedlaunch" };
+    static LPCWSTR c_contractIdKeyName{ L"ContractId" };
+
+    // This interface used for the internal value marshaling mechanism.  Implementing this interface means
+    // that both Serialize and Deserialize be implemented.  Deserialize is a static and can't be expressed
+    // here.  Deserialize is required also by the encoded launch mechanism, and is how ActivatedEventArgs
+    // are created regardless of marshaling.  The below method is only used for marshaling.
+    MIDL_INTERFACE("1E875DD8-3E09-465D-8684-E76EF58145C5") IInternalValueMarshalable : IInspectable
+    {
+        virtual winrt::Windows::Foundation::Uri Serialize() PURE;
+    };
+
+    inline std::wstring GenerateEncodedLaunchUri(std::wstring const& appUserModelId, std::wstring const& contractId)
```

Comment:
an example (comment) of what this looks like "in real life" is priceless. :-)
L"%s://%s?%s=%s"